### ISSUE TITLE
Gen 7 Egg Accept/Reject Calculation

### ISSUE THREAD

**Issue by Admiral-Fish** [zaksabeast/SMBrowserRNG#45]
Should probably include the secondary egg view to focus on accept/reject calculations.

https://gyazo.com/9105309682353826ef9ebc85b7a6430e

Also should consider using Dijkstra's formula to calculate shortest path of accept/rejects.

https://www.reddit.com/r/pokemontrades/comments/61qhe6/daily_discussion_question_thread_for_27_march_2017/dfgj9uo/

**Comment by zaksabeast** [comment]
The formula looks like it could work well for this.  Here's a node version - https://www.npmjs.com/package/node-dijkstra

I don't think we'll need a new view for this - we can just add a new setting box.

**Comment by pkmnbumba** [comment]
I talked to fish about this but not completely sure if this is the best thing to implement (Dijkstra's Algorithm). If it is implemented here, I'd highly suggest using a checkbox for whether the user wants it cause I think it'd have a pretty long runtime. Doesn't hurt to try though :p

**Comment by Admiral-Fish** [comment]
Yeah a checkbox was the original idea I had in mind. User could choose between default or fastest path.

**Comment by pkmnbumba** [comment]
It could also come as a button AFTER the original search (for example, Calculate Shortest Path). This would probably make the most sense since you would need all the data from the entire search anyways to implement a shortest paths algorithm.

**Comment by zaksabeast** [comment]
I really do like the Dijkstra method, however this does require the original data first, so a checkbox makes sense.

For the user it makes sense as well - they get to see the results before deciding which egg to get the shortest path for.

Having the button be usable after the search I like, however I think it should only be disabled until a search is performed, not invisible - users know the option is there in that way.

What are everyone's thoughts on a disabled button vs. invisible button, as well as if we should use Dijkstra's Algorithm or a different method?

**Comment by pkmnbumba** [comment]
Disabled button sounds best to me, otherwise it may be easily overlooked.

I was reading law's method thats described in fish's reddit post but I'm not thoroughly convinced, at least not the last 10 accepts. I don't think that would affect the # of accepts/rejects enough to make it worth it. I would say dijkstra's, especially if its already available as a package. We might need to do some testing though - I like the idea of using the algorithm for the last X number of accepts. We probably need to find the sweet point that hits a decent runtime but also actually saves time. (honestly, I'm thinking more of its application to shiny TSV RNGs, I think dijkstra's will run fine for normal breedables)
